### PR TITLE
[Dashboard] Definitions and Implementations of Tenant Management API

### PR DESCRIPTION
**Pull request by PowerfooI**

### CODE REVIEW

File: distribution/dashboard/internal/router/v1/obtenant_router.go [hunk 1]
```diff
@@ -9,6 +9,19 @@ func InitOBTenantRoutes(g *gin.RouterGroup) {
...
+
+	g.POST("/obtenant/:namespace/:name/logreplay", handler.ReplayStandbyLog)
+	g.POST("/obtenant/:namespace/:name/version", handler.UpgradeTenantVersion)
+	g.POST("/obtenant/:namespace/:name/role", handler.ChangeTenantRole)
+
+	g.GET("/obtenant/:namespace/:name/backupPolicy", handler.GetBackupPolicy)
+	g.PUT("/obtenant/:namespace/:name/backupPolicy", handler.CreateBackupPolicy)
+	g.POST("/obtenant/:namespace/:name/backupPolicy", handler.UpdateBackupPolicy)
+	g.DELETE("/obtenant/:namespace/:name/backupPolicy", handler.DeleteBackupPolicy)
+	g.GET("/obtenant/:namespace/:name/:type/backupJobs", handler.ListBackupJobs)
```
Comment (chris-sun-star) [inline distribution/dashboard/internal/router/v1/obtenant_router.go:26]: maybe this is more reasonable /obtenant/:namespace/:name/backup/:type/jobs

File: distribution/dashboard/internal/router/v1/obtenant_router.go [hunk 2]
```diff
@@ -9,6 +9,19 @@ func InitOBTenantRoutes(g *gin.RouterGroup) {
 	g.GET("/obtenants", handler.ListAllTenants)
 	g.GET("/obtenant/:namespace/:name", handler.GetTenant)
 	g.PUT("/obtenant/:namespace/:name", handler.CreateTenant)
-	g.POST("/obtenant/:namespace/:name", handler.UpdateTenant)
 	g.DELETE("/obtenant/:namespace/:name", handler.DeleteTenant)
+
+	g.PUT("/obtenant/:namespace/:name/unitNumber", handler.ModifyUnitNumber)
```
Comment (chris-sun-star) [inline distribution/dashboard/internal/router/v1/obtenant_router.go:14]: why not using patch for modification
ob-operator supports modify multiple configurations

File: distribution/dashboard/internal/model/response/obtenant.go [hunk 3]
```diff
@@ -1,12 +1,49 @@
 package response
 
-import "github.com/oceanbase/ob-operator/api/v1alpha1"
+// @Description Brief information about OBTenant
+type OBTenantBrief struct {
+	Name        string            `json:"name"`        // Name of the resource
+	Namespace   string            `json:"namespace"`   // Namespace of the resource
+	TenantName  string            `json:"tenantName"`  // Name of the tenant in the database
+	ClusterName string            `json:"clusterName"` // Name of the cluster belonging to
```
Comment (chris-sun-star) [inline distribution/dashboard/internal/model/response/obtenant.go:8]: is this ClusterName the CR name or the cluster name in OceanBase

Comment (PowerfooI) [reply distribution/dashboard/internal/model/response/obtenant.go:8]: This is name of CR. Tenant would not feel cluster name in the OceanBase.

File: distribution/dashboard/internal/router/v1/obtenant_router.go [hunk 4]
```diff
@@ -9,6 +9,17 @@ func InitOBTenantRoutes(g *gin.RouterGroup) {
 	g.GET("/obtenants", handler.ListAllTenants)
 	g.GET("/obtenant/:namespace/:name", handler.GetTenant)
 	g.PUT("/obtenant/:namespace/:name", handler.CreateTenant)
-	g.POST("/obtenant/:namespace/:name", handler.UpdateTenant)
 	g.DELETE("/obtenant/:namespace/:name", handler.DeleteTenant)
+	g.PATCH("/obtenant/:namespace/:name", handler.PatchTenant)
+
+	g.POST("/obtenant/:namespace/:name/rootPassword", handler.ChangeRootPassword)
```
Comment (chris-sun-star) [inline distribution/dashboard/internal/router/v1/obtenant_router.go:15]: it's better to use credential in path and support other users, pass the username and password in request body

Comment (PowerfooI) [reply distribution/dashboard/internal/router/v1/obtenant_router.go]: OK, I'll rewrite it to a more generic one.

**Review by chris-sun-star — APPROVED** [review]

**Pull request merged**